tests/simpleKVBC/TesterReplica/setup.cpp

```diff
@@ -243,4 +270,50 @@ std::unique_ptr<IStorageFactory> TestSetup::GetStorageFactory() {
   return std::make_unique<v2MerkleTree::RocksDBStorageFactory>(dbPath.str());
 }
 
+std::vector<std::string> getKeyDirectories(const std::string& keysRootPath)
+{
+  std::vector<std::string> result;
+  std::string fullKeyPath = keysRootPath + "/transaction_signing_keys";
+  for(auto& res : std::filesystem::recursive_directory_iterator(fullKeyPath)) {
+    if (res.is_directory()) {
+      result.push_back(res.path().string());
+    }
+  }
+  return result;
+}
+
+void setPublicKeysOfClients(const std::string& principalsMapping,
+                            const std::string& keysRootPath,std::map<const std::string,
+                            std::set<const uint16_t>>& publicKeysOfClients) {
+  std::vector<std::string> keysDirectories = getKeyDirectories(keysRootPath);
+  std::vector<std::string> clientIdChunks;
+  boost::split(clientIdChunks, principalsMapping, boost::is_any_of(","));
+
+  if ( clientIdChunks.size() != keysDirectories.size() ) {
+    throw std::runtime_error("Number of keys directory should match the number of number of sets of clientIds mappings in principals mapping string");
+  }
+
+  // Sort keysDirectories just to ensure ordering of the folders
+  std::sort(keysDirectories.begin(), keysDirectories.end());
+
+  // Build publicKeysOfClients of replicaConfig
+  for(int i = 0; i < clientIdChunks.size(); ++i) {
+    std::string keyFilePath = keysDirectories.at(i) + "/public_key.pem"; // Will be "../1/public_key.pem" etc.
+
+    SecretsManagerPlain smp;
+    std::optional<std::string> keyPlaintext;
+    keyPlaintext = smp.decryptFile(keyFilePath);
+    if (keyPlaintext.has_value()) {
+      // Each clientIdChunk would look like "11 12 13 14"
+      std::vector<std::string> clientIds;
+      boost::split(clientIds, clientIdChunks.at(i), boost::is_any_of(" "));
```

Comment:
2 comments:
1) the next code will work, why use a vector?
`
    std::set<std::string> clientIds;
    boost::split(clientIds, "11 12 13", boost::is_any_of(" "));
`

2) Definition of publicKeysOfClients:
`std::set<std::pair<const std::string, std::set<uint16_t>>> publicKeysOfClients;`

Sorry for the bad sync.
That means, you need to transform this set into another set of uint16_t

Reply:
I think boost::split expects vector input, so I am collecting it as a vector and then converting it into a set. 
No your sync was fine, it was a mistake on my end. The set needed `uint16_t` not string. I have corrected that.

Reply:
boost::split works with other containers as well..
This code compiles and work:

std::set<std::string> clientIds; 
boost::split(clientIds, "11 12 13", boost::is_any_of(" "));

Reply:
I see. However, it will result in a collection of string, which we will again have to convert into a set of `uint16_t`. So I am wondering that we just leave the vector as is, and add the string to int conversion in a loop. Here's the updated code:

```
      std::vector<std::string> clientIds;
      boost::split(clientIds, clientIdChunks.at(i), boost::is_any_of(" "));
      std::set<uint16_t> clientIdsSet;
      for (const std::string clientId : clientIds) {
        clientIdsSet.insert(std::stoi(clientId));
      }
```

Reply:
Yeah, I guess it's fine that way :)